internal/authtoken/repository.go

```diff
@@ -91,10 +89,9 @@ func (r *Repository) CreateAuthToken(ctx context.Context, withIamUser *iam.User,
 		return nil, fmt.Errorf("create: unable to get database wrapper: %w", err)
 	}
 
-	// TODO: Allow the caller to specify something different than the default duration.
 	// We truncate the expiration time to the nearest second to make testing in different platforms with
 	// different time resolutions easier.
-	expiration, err := ptypes.TimestampProto(time.Now().Add(maxTokenDuration).Truncate(time.Second))
+	expiration, err := ptypes.TimestampProto(time.Now().Add(globals.DefaultAuthTokenMaxDuration).Truncate(time.Second))
```

Comment:
For testing purposes it's much better if this is simply a property of the controller (via its RawConfig) that can then be passed through to any auth methods via the auth service handler. So rather than have globals, when the controller constructs the auth method service handler, it should provide the values from its configuration.

This does mean updating quite a bit more code elsewhere, although one way around this (and likely better overall) is to add an Option capability to the constructors for the service handler and the repositories that allow this to be passed through, or not.

Reply:
This was something I wasn't sure of: use a global or pass it in as an option. I went with the former because of brevity, but I think the appropriate pattern to set is using a functional option on the auth token type which is what _I think_ you're pointing to. A bit more code wrangling but a better long term pattern, does that sound right?

Reply:
Yep, exactly -- more code now but better long term pattern. As the comment in globals.go says, globals are bad, and we can definitely avoid them in this case. However, globals can provide the default! So feel free to keep values there, just use that as a default for if config values aren't specified -- or maybe even better, move them to the config package so they live next to the parsing code. But that way you can always be sure that a config's value will end up with some concrete >0 value regardless of it it was the default or not.

Reply:
Fixed